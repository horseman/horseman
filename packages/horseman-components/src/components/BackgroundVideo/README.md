```
const data = {
  src: "/path/to/video.mp4",
  type: "video/mp4",
};

<BackgroundVideo {...data}>
Content Goes Here
</BackgroundVideo>
```
