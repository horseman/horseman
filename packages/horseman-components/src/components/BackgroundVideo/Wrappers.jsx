import styled from "styled-components";

export const ComponentWrapper = styled.div`
  position: relative;
  overflow: hidden;
`;

export const Content = styled.div`
  position: relative;
  z-index: 1;
`;

export const Video = styled.video.attrs({
  autoPlay: true,
  muted: true,
  loop: true,
  playsInline: true,
})`
  position: absolute;
  top: 0;
  left: 0;
  min-width: 100%;
  z-index: 0;
  width: 100vw;

  height: auto;
  background: white;
  background-position: center center;
  background-size: cover;
  display: block;
`;
