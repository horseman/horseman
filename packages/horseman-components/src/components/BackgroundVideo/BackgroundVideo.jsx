import React from "react";
import PropTypes from "prop-types";

import { ComponentWrapper, Content, Video } from "./Wrappers";

const BackgroundVideo = ({ children, ...rest }) => (
  <ComponentWrapper>
    <Video>
      <source {...rest} />
    </Video>
    <Content>{children}</Content>
  </ComponentWrapper>
);

BackgroundVideo.propTypes = {
  children: PropTypes.node.isRequired,
  video: PropTypes.shape({
    src: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
};

export default BackgroundVideo;
