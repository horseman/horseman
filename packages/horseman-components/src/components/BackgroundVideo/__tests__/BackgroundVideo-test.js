import React from "react";
import { shallow } from "enzyme";

import BackgroundVideo from "../";
import { Video, Content } from "../Wrappers";

describe("BackgroundVideo", () => {
  test("renders correctly", () => {
    const wrapper = shallow(
      <BackgroundVideo src="foo" type="bar">
        foo
      </BackgroundVideo>,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
  test("Video", () => {
    const wrapper = shallow(<Video />);
    expect(wrapper).toMatchSnapshot();
  });
  test("Content", () => {
    const wrapper = shallow(<Content />);
    expect(wrapper).toMatchSnapshot();
  });
});
