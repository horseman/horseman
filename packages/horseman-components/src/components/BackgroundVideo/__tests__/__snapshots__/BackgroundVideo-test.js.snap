// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundVideo Content 1`] = `
.c0 {
  position: relative;
  z-index: 1;
}

<div
  className="c0"
/>
`;

exports[`BackgroundVideo Video 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  min-width: 100%;
  z-index: 0;
  width: 100vw;
  height: auto;
  background: white;
  background-position: center center;
  background-size: cover;
  display: block;
}

<video
  autoPlay={true}
  className="c0"
  loop={true}
  muted={true}
  playsInline={true}
/>
`;

exports[`BackgroundVideo renders correctly 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
}

<div
  className="c0"
>
  <styled.video>
    <source
      src="foo"
      type="bar"
    />
  </styled.video>
  <styled.div>
    foo
  </styled.div>
</div>
`;
